include/api.php

```diff
@@ -3629,6 +3629,71 @@ function api_direct_messages_destroy($type)
 /// @TODO move to top of file or somewhere better
 api_register_func('api/direct_messages/destroy', 'api_direct_messages_destroy', true, API_METHOD_DELETE);
 
+function api_friendships_destroy($type)
+{
+        $a = get_app();
+
+        if (api_user() === false) {
+                throw new ForbiddenException();
+        }
+	$uid = local_user();
+
+        $contact_id = (x($_REQUEST, 'user_id') ? $_REQUEST['user_id'] : null);
+
+        if ($contact_id == null) {
+		logger("No given user_id", LOGGER_DEBUG);
+                throw new BadRequestException("no user_id specified");
+        }
+
+	// Get Contact by given id
+        $contact = q("SELECT * FROM `contact` WHERE `id` = %d AND `uid` = %d AND `self` = 0", intval($contact_id), 0);
+
+	if(!DBA::isResult($contact)) {
+		logger("No contact by id founded", LOGGER_DEBUG);
+		throw new BadRequestException("no contact found to given ID");
+	}
+
+	$url = $contact[0]["url"];
```

Comment:
When you convert the above `q()` call to `DBA::selectFirst()`, this should be simplified to `$url = $contact["url"];`